### PR TITLE
Enable in-game changes for pad settings

### PR DESCRIPTION
**Pull request by Megamouse**

### CODE REVIEW

File: rpcs3/pad_thread.h [hunk 1]
```diff
@@ -35,7 +37,8 @@ class pad_thread
 	PadInfo m_info;
 	std::vector<std::shared_ptr<Pad>> m_pads;
 
-	bool active;
+	bool active{ false };
+	bool reset{ false };
```
Comment (elad335) [inline rpcs3/pad_thread.h]: I know it was already like this but, can you make these atomic?

File: rpcs3/rpcs3_app.cpp [hunk 2]
```diff
@@ -142,6 +142,15 @@ void rpcs3_app::InitializeCallbacks()
 		RequestCallAfter(std::move(func));
 	};
 
+	callbacks.reset_pads = [this]()
+	{
+		const auto pad_handler = pad::get_current_handler();
+		if (pad_handler)
```
Comment (elad335) [inline rpcs3/rpcs3_app.cpp]: Can't be null.

**Comment by vsub** [comment]
Nice,this will finally fix my problem.
My gamepad is registered twice(wireless(the dongle is always connected and registered as XInput\MMJoystic 0 or 1)and wired(registered as what wired mode is not set))and I can't switch modes mid game

File: rpcs3/rpcs3_app.cpp [hunk 3]
```diff
@@ -142,6 +142,12 @@ void rpcs3_app::InitializeCallbacks()
 		RequestCallAfter(std::move(func));
 	};
 
+	callbacks.reset_pads = [this]()
+	{
+		std::lock_guard lock(pad::g_pad_mutex);
+		pad::get_current_handler()->Reset();
```
[the remark this answers is not in the document]
Comment (elad335) [reply rpcs3/rpcs3_app.cpp]: When I said "before every get_handler" I meant before the actual usage of it.

[the remark this answers is not in the document]
Comment (Megamouse) [reply rpcs3/rpcs3_app.cpp]: So what. It's not doing any harm :100:

**Pull request merged**